File: YABA-New/YABA/app/src/main/java/dev/subfly/yaba/core/state/creation/notemark/NotemarkMentionCreationEvent.kt
package dev.subfly.yaba.core.state.creation.notemark

sealed interface NotemarkMentionCreationEvent {
    data class OnInit(
        val initialText: String = "",
        val initialBookmarkId: String? = null,
        val isEdit: Boolean = false,
        val editPos: Int? = null,
    ) : NotemarkMentionCreationEvent

    data class OnChangeMentionText(
        val text: String,
    ) : NotemarkMentionCreationEvent

    /**
     * User finished the bookmark picker; [bookmarkId] is the only payload from selection.
     * Mention text is auto-filled from the bookmark label when it is still blank.
     */
    data class OnBookmarkPickedFromSelection(
        val bookmarkId: String,
    ) : NotemarkMentionCreationEvent
}

File: Darwin/YABACore/State/Creation/Notemark/NotemarkMentionCreationEvent.swift
//
//  NotemarkMentionCreationEvent.swift
//  YABACore
//
//  Parity with Compose `NotemarkMentionCreationEvent` (creation/notemark).
//

import Foundation

public enum NotemarkMentionCreationEvent: Sendable {
    case onInit(
        initialText: String,
        initialBookmarkId: String?,
        isEdit: Bool,
        editPos: Int?
    )
    case onChangeMentionText(String)
    /// User finished bookmark selection; host may supply `bookmarkLabel` so mention text can auto-fill when still blank.
    case onBookmarkPickedFromSelection(bookmarkId: String, bookmarkLabel: String?)
}
